scripts/standardise_validate.py

```diff
@@ -1,14 +1,58 @@
 import argparse
 import json
 import os
+from typing import List, Tuple
 
 from linz_logger import get_log
 
 from scripts.cli.cli_helper import format_date, format_source, is_argo, valid_date
 from scripts.create_stac import create_item
-from scripts.files.fs import write
+from scripts.files.file_tiff import FileTiff
+from scripts.files.files_helper import get_file_name_from_path, is_tiff, is_vrt
+from scripts.files.fs import exists, read, write
 from scripts.gdal.gdal_helper import get_srs, get_vfs_path
+from scripts.gdal.gdalinfo import gdal_info
 from scripts.standardising import run_standardising
+from scripts.tile.tile_index import Point, get_tile_name
+
+
+def get_processed_tiffs(source: List[str], target: str, scale: int) -> Tuple[List[str], List[FileTiff]]:
+    """By checking the target path, return a list of the source path that have been processed
+    and a list of tiff that haven't a STAC file associated.
+
+    Args:
+        source: the list of image source path
+        target: the target location for the processed output (tiff and stac)
+        scale: the scale of the dataset
+
+    Returns:
+        A tuple containing processed source path and a list of tiff with no stac
+    """
+    processed_tiffs: List[str] = []
+    no_stac: List[FileTiff] = []
+    for path in source:
+        if is_tiff(path) or is_vrt(path):
+            # gdalinfo the original file
+            gdalinfo = gdal_info(path)
+            origin = Point(gdalinfo["cornerCoordinates"]["upperLeft"][0], gdalinfo["cornerCoordinates"]["upperLeft"][1])
+            tile_name = get_tile_name(origin, scale)
+            prefix = os.path.join(target, tile_name)
+            target_path_stac = prefix + ".json"
+            target_path_tiff = prefix + ".tiff"
+            # STAC item json file name
+            if exists(target_path_stac):
+                # STAC exists, consider the TIFF exists too
+                processed_tiffs.append(path)
```

Comment:
it should validate that the tiff exists too. It's only complete if both are present.